### ISSUE TITLE
[Feature Suggestion] Add ROF choice to full auto

### ISSUE THREAD

**Issue by Paulkwk** [OctarineSourcerer/cyberpunk2020#143]
![image](https://github.com/OctarineSourcerer/cyberpunk2020/assets/170223593/bed48e54-3c3a-4242-80af-4e15fb375c28)
The full auto rules indicate that if using full auto on a different target, you need to split the ROF by the number of targets. 
It means that rof will change according to how many targets is getting shot  at. 
I suggest a fillable box like a modifier, in the attack popup. So people can choose how many round is used for this target. 

Thank you so much for your work.

**Comment by OctarineSourcerer** [comment]
Right, so I had some time today, and should be in c95fa4efd8e8fde0ffc5ef4d6c6a230ea0742d25! Though it's certainly not well-rounded enough to officially release. This is something I intended to get working in some form in any case, as the original intended experience is "select targets, fire, everything happens for you".

You can now select token targets before firing, via Foundry's usual controls - you can either click a token then its bullseye, or you can hover over a token and press T. Press shift-T to ADD a token to whichever ones are already selected.

The ranged weapon dialog should now list targets you have selected, if you've selected any. If you _haven't_, there's still a "targets count" modifier on the dialog now. That's editable if you haven't selected any targets, but locked if you have - it'll draw it entirely from the number of targets you've selected. (_actually, in that case, I should probably have that modifier disappear so it doesn't get confusing!_).

When you go to roll full-auto specifically, it'll roll separate autofires for each target. Be warned at the moment that if selected tokens give the same name, it'll list the same name, so it'll be hard to figure out which of them to modify. It'll be a good idea for clicking that name to bring up the appropriate token - or simply a button to apply damage to that token - but that's not in yet.

Other flaws right now are that it'll use the range you give in the modifiers (instead of drawing it individually from each target token). Non-autofire modes do not use these targets currently, and the UI doesn't make it clear when target information _isn't_ used.

Oh, and ammunition is now actually removed when firing ranged weapons! That is likely to trip some people up, as currently we don't make clear how much ammo remains in a weapon.

In any case though, that commit should give you something to start with.

**Comment by Paulkwk** [comment]
> Right, so I had some time today, and should be in [c95fa4e](https://github.com/OctarineSourcerer/cyberpunk2020/commit/c95fa4efd8e8fde0ffc5ef4d6c6a230ea0742d25)! Though it's certainly not well-rounded enough to officially release. This is something I intended to get working in some form, as the original intended experience was "select targets, fire, everything happens for you".
> 

wow, this is quick, thank you so much, this is amazing. 
but…
I think making it automatic will be much harder to implement then you expected. Since according to the “listening up you primitive screw head” book, each empty space between the targets also counts as targets….

**Comment by OctarineSourcerer** [comment]
> > Right, so I had some time today, and should be in [c95fa4e](https://github.com/OctarineSourcerer/cyberpunk2020/commit/c95fa4efd8e8fde0ffc5ef4d6c6a230ea0742d25)! Though it's certainly not well-rounded enough to officially release. This is something I intended to get working in some form, as the original intended experience was "select targets, fire, everything happens for you".
> 
> wow, this is quick, thank you so much, this is amazing. but… I think making it automatic will be much harder to implement then you expected. Since according to the “listening up you primitive screw head” book, each empty space between the targets also counts as targets….

That doesn't surprise me in the slightest, though I don't have the primitive screwhead book myself; could you put a screenshot of the relevant part in here?

And yes, it'll almost certainly involve nudging Foundry's canvas for a few things if so, that'll be some new territory...

**Comment by Paulkwk** [comment]
> > > Right, so I had some time today, and should be in [c95fa4e](https://github.com/OctarineSourcerer/cyberpunk2020/commit/c95fa4efd8e8fde0ffc5ef4d6c6a230ea0742d25)! Though it's certainly not well-rounded enough to officially release. This is something I intended to get working in some form, as the original intended experience was "select targets, fire, everything happens for you".
> > 
> > 
> > wow, this is quick, thank you so much, this is amazing. but… I think making it automatic will be much harder to implement then you expected. Since according to the “listening up you primitive screw head” book, each empty space between the targets also counts as targets….
> 
> That doesn't surprise me in the slightest, though I don't have the primitive screwhead book myself; could you put a screenshot of the relevant part in here?
> 
> And yes, it'll almost certainly involve nudging Foundry's canvas for a few things if so, that'll be some new territory...

Ah, I'm sorry, after going through the book, I can't find where it is. But I remember seeing it somewhere, maybe in other books...
 I think giving a fillable slot that lets players override the ROF will be the easiest option, and also the most versatile.
